pyms/flask/app/create_app.py

```diff
@@ -192,11 +201,32 @@ def init_metrics(self) -> None:
         if self.metrics:
             self.application.register_blueprint(self.metrics.metrics_blueprint)
             self.metrics.add_logger_handler(
-                self.application.logger,
-                self.application.config["APP_NAME"]
+                self.application.logger, self.application.config["APP_NAME"]
             )
             self.metrics.monitor(self.application.config["APP_NAME"], self.application)
 
+    def init_opentelemetry(self) -> None:
+        if self.opentelemetry:
+            if self.opentelemetry.config.metrics.enabled:
+                # Set metrics backend
+                self.opentelemetry.set_metrics_backend()
+                # Set the metrics blueprint
+                # DISCLAIMER this endpoint may be only necessary with prometheus client
+                self.application.register_blueprint(self.opentelemetry.blueprint)
+                # Set instrumentations
+                if self.opentelemetry.config.metrics.instrumentations.flask:
+                    self.opentelemetry.monitor(
+                        self.application.config["APP_NAME"], self.application
+                    )
+                if self.opentelemetry.config.metrics.instrumentations.logger:
+                    self.opentelemetry.add_logger_handler(
+                        self.application.logger, self.application.config["APP_NAME"]
+                    )
+            if self.opentelemetry.config.tracing.enabled:
```

Comment:
@avara1986 Do you think it make sense to add this to the PR? They won't be useful, since the tracing and logging are out of scope, but it may be useful to have it already designed.

Reply:
with the new method of serviceDriver `def init_action(self, microservice_instance):` could be more easy to read the other services. But we need to check other way to check the services that add `if service`, `if service`, `if service`... ;)

Reply:
Open telemetry is a kind of a meta service. As it may instrument other services (there is an issue for redis integration already afair) it seems that configuring integration should actually be a responsibility of the service that provides instrumented module. Maybe in similar style to ‘init_action’ there should be ‘init_telemetry’ that would be called by opentelemetry service? Or is it too tightly coupled?
There is just no way to generically initialise integrations. Don’t know why did they do it like that. For example sentry sdk just gets a list of integrations to enable - most (if not all) of those work well without any params.

Reply:
What you're proposing that for every service there's a telemetry section? It may make sense, but not all possible instrumentation of OT will be available as service. 
Also, maybe you don't want to use the service that PyMS provides (for example requests) but you do want OT for requests. 
Are you talking about this?

Reply:
I've checked code for some connectors that would be useful for services (not only documentation this time) and while some of example instrumentations take params it seems to really be just an optional tracer. So it's possible to configure those by simply iterating over a list after all. There aren't that many instrumentations that need custom setup and would be useful here - Celery, Flask, WSGI, maybe a few more that may work here due to not being async-related. But requests also uses optional callback so advanced config may be useful here. Also not sure if support for multiple tracers is important?
But still find per-service telemetry section appealing - instead of something this:
```yaml
opentelemetry:
      integrations:
      - redis
      - requests
```
you end with something like this:
```yaml
redis:
      telemetry: true
requests:
      telemetry: true
```

About `you don't want to use the service but you do want OT` I'd argue that if you're not using the provided service then instrumenting this part is on you - either in `init_libs` or new app-level `init_telemetry` - as it may be necessary to enable instrumentation before lib usage. Alternatively it can be done like this:
```yaml
redis:
      telemetry: true
      service_enabled: false
requests:
      telemetry: true
```
But there is one major advantage when using OT instrumentation list in config: those simple instrumentations may be enabled completely without a service - or even without any additional support in pyms. But then something like this would need to be suported:
opentelemetry:
      integrations:
      - opentelemetry.instrumentation.redis.RedisInstrumentor
```